### PR TITLE
Document stub databases

### PR DESCRIPTION
**Pull request by melmothx**

### CODE REVIEW

File: concepts/indexing/databases.rst [hunk 1]
```diff
@@ -62,11 +62,76 @@ database when the physical database location may vary; for example, because
 a new database is being built nightly, and is named according to the date
 on which it was built.
 
-.. todo:: document format, or link to documentation of it
+These files are recognised by the autodetection in the Database
+constructor (if the pathname is file rather than a directory, it's
+treated as a stub database file, or if the pathname is a directory
+containing a file called ``XAPIANDB``) or you can open them explicitly
```
[the remark this answers is not in the document]
Comment (ojwb) [reply concepts/indexing/databases.rst]: Yes, a single-file database is just a file.

Xapian can usually tell between a single-file database and a stub database file by looking at the file size - a stub database file is usually very small, while a single-file database is at least 2KB and (for glass at least) a multiple of 2KB in size.  So if it's not a multiple of 2KB we just open it as a stub; if it is, then it's almost certain a single-file database but we need to look at the start of the file to see if it has the magic sequence which a single-file database has.  This is probably not something that's worth documenting here, just background information really.

[the remark this answers is not in the document]
Comment (jaylett) [reply concepts/indexing/databases.rst]: Maybe we should just drop everything in parentheses?

[the remark this answers is not in the document]
Comment (ojwb) [reply concepts/indexing/databases.rst]: That sounds good to me.  We should probably still mention the `XAPIANDB` feature somewhere though.  Richard added it mostly for replication, but it's probably useful in other situations.

[the remark this answers is not in the document]
Comment (melmothx) [reply concepts/indexing/databases.rst]: Where the `XAPIANDB` feauture is documented? IIUC, if you pass a directory, and the directory has `XAPIANDB`, this file would be the stub database?

**Comment by ojwb** [comment]
> Where the XAPIANDB feauture is documented? IIUC, if you pass a directory,
> and the directory has XAPIANDB, this file would be the stub database?.
>
Yes and documented here: https://xapian.org/docs/overview.html

**Comment by melmothx** [comment]
Olly Betts <notifications@github.com> writes:

>> Where the XAPIANDB feauture is documented? IIUC, if you pass a directory,
>> and the directory has XAPIANDB, this file would be the stub database?.
>>
> Yes and documented here: https://xapian.org/docs/overview.html

Ok (Wasn't really crystal clear).


-- 
Marco

**Comment by ojwb** [comment]
This looks good to me.

@jaylett You requested changes, are you happy those comments have been addressed?

**Pull request merged**